Tutorials/intro-to-octave/configuration.py

```diff
@@ -56,127 +40,151 @@
             "intermediate_frequency": IF,
             "operations": {
                 "cw": "const",
+                "readout": "readout_pulse",
             },
             "digitalInputs": {
                 "switch": {
-                    "port": ("con1", 4),
+                    "port": (con, 7),
                     "delay": 136,
                     "buffer": 0,
                 },
             },
         },
         "qe5": {
-            "mixInputs": {
-                "I": ("con1", 9),
-                "Q": ("con1", 10),
-                "lo_frequency": LO,
-                "mixer": "octave_octave1_5",  # a fixed name, do not change.
-            },
+            'RF_inputs': {"port": (octave, 5)},
             "intermediate_frequency": IF,
             "operations": {
                 "cw": "const",
+                "readout": "readout_pulse",
             },
             "digitalInputs": {
                 "switch": {
-                    "port": ("con1", 5),
+                    "port": (con, 9),
                     "delay": 136,
                     "buffer": 0,
                 },
             },
         },
+
+    },
+    "octaves": {
+        octave: {
+            "RF_outputs": {
+                1: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",  # can be external or internal. internal is the default
+                    "output_mode": "always_on", # can be: "always_on" / "always_off"/ "triggered" / "triggered_reversed". "always_off" is the default
+                    "gain": 0, # can be in the range [-20 : 0.5 : 20]dB
+                },
+                2: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                3: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                4: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                5: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+            },
+            "RF_inputs": {
+                1: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal", # internal is the default
+                    "IF_mode_I": "direct",  # can be: "direct" / "mixer" / "envelope" / "off". direct is default
+                    "IF_mode_Q": "direct",
+                },
+                2: {
+                    "LO_frequency": LO,
+                    "LO_source": "external", # external is the default
```

Comment:
Just to double check that I understood, here the Octave knows which LO port is connected to the dmd2LO by the RF_input key, right? This means that is for qe2, I set RF_input=3, then the Octave will know that LO3 is connected to dmd2LO?

Reply:
Hey, it doesn't know which LO port (from the back panel) is connected to which DmdLO is connected to in the back panel. Here it's just external LO for him... There is a loopback key for this purpose. For example:

    "octaves": {
        "octave1": {
            "RF_outputs": {...},
            "RF_inputs": {...},
            "loopbacks": [
                (("octave1", 'Synth2'), 'Dmd2LO')
            ],
            "connectivity": "con1"
        }
    },